hazelcast/src/main/java/com/hazelcast/cp/internal/client/AddCPGroupAvailabilityListenerMessageTask.java

```diff
@@ -53,7 +53,7 @@ protected CompletableFuture<UUID> processInternal() {
 
         boolean local = parameters;
```

Comment:
Possibly we can publish a "global" or "async" event only in the case of a graceful shutdown.

Reply:
Not sure that's possible -- that's coming from the client; there's no way I see to distinguish from that context. Could be wrong...

Reply:
`ClientListenerServiceImpl#registerseLocalOnly()` https://github.com/hazelcast/hazelcast/blob/f31272b3c21b991411cdb3b6fe78d6cec6faf2fa/hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerServiceImpl.java#L307 determines the value of `local`.

So, this is true so long as the smart routing is not disabled. https://github.com/hazelcast/hazelcast/blob/f31272b3c21b991411cdb3b6fe78d6cec6faf2fa/hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerServiceImpl.java#L90

Reply:
See my above comment on just using a single semantics for this and effectively making `local` surplus.